code/game/gamemodes/dynamic/dynamic.dm

```diff
@@ -506,7 +506,7 @@ GLOBAL_VAR_INIT(dynamic_forced_threat_level, -1)
 		if (initial(ruleset_type.weight) == 0)
 			continue
 
-		var/ruleset = new ruleset_type
```

Comment:
instead of adding a new snowflake constructor argument, you can directly do `ruleset.mode = src`. Nothing else uses this behavior so imo it's kinda of bad to pass in for a special case like this

Reply:
The point of a constructor argument is when it is a mandatory field. If it is not passed in during constructor time, then it will fail. Having this in the constructor is better because it shows to the programmer that they are required to pass in this parameter. Passing in required parameters into the constructor is seen as good programming practice and discourages accidental misuse.

Reply:
This is the non-simulation instantiation. /datum/dynamic_ruleset requires mode to be passed in regardless of whether or not it was used in a simulation.
You are right in that I missed some instantiations, I missed the ones inside of the admin UI but I don't think that would cause issues. I will update them regardless. If only byond was well typed and could catch this mistake